timemachine/fe/stored_arrays.py

```diff
@@ -0,0 +1,140 @@
+import io
+import shutil
+import tempfile
+import weakref
+from itertools import count
+from pathlib import Path
+from typing import Iterator, List, NoReturn, Sequence, overload
+
+import numpy as np
+from numpy.typing import ArrayLike, NDArray
+
+from timemachine.parallel.client import AbstractFileClient
+
+
+class StoredArrays(Sequence[NDArray]):
+    """Sequence of numpy arrays using O(1) memory, backed by disk storage.
+
+    Data is stored in a temporary directory that is cleaned when the `StoredArrays` object is finalized.
+
+    Examples
+    --------
+    >>> sa = StoredArrays()
+    >>> sa.extend([np.array([1, 2, 3]), np.array([4, 5, 6])])
+    >>> len(sa)
+    2
+    >>> len(list(sa._chunks()))
```

Comment:
Should `_chunks` be a non-private function?

Reply:
Hmm, I don't think so -- it's included in the example here for documentation purposes (to give some intuition about how arrays are stored internally). I'd prefer to remove this example if it seems confusing.

Reply:
Documenting private functions seem weird to me, would rather just document `_chunks` if the intent is to clarify how things are stored.

Reply:
Done in https://github.com/proteneer/timemachine/pull/950/commits/d529f3105732308538cbb2d3e6caf4f94d8ce466